### PR TITLE
Throw exception in RPC

### PR DESCRIPTION
**Pull request by KaustubhShamshery**
## Description
For community devs, make it easier to see why CreateTransaction fails

### Implementation

- [x] **ready for review**

### Integration Test (Core Team)

- [ ] local machine test

- [ ] small-scale cloud test

### CODE REVIEW

File: src/libServer/JSONConversion.cpp [hunk 1]
```diff
@@ -139,7 +139,8 @@ const Transaction JSONConversion::convertJsontoTx(const Json::Value& _json) {
   bytes toAddr_ser;
   if (!DataConversion::HexStrToUint8Vec(lower_case_addr, toAddr_ser)) {
     LOG_GENERAL(WARNING, "json cointaining invalid hex str for toAddr");
```
Comment (evesnow91) [inline src/libServer/JSONConversion.cpp]: "cointaining" -> containing

File: src/libServer/JSONConversion.cpp [hunk 2]
```diff
@@ -196,42 +199,51 @@ bool JSONConversion::checkJsonTx(const Json::Value& _json) {
...
       LOG_GENERAL(INFO,
                   "To Address checksum wrong " << _json["toAddr"].asString());
-      return false;
+      throw jsonrpc::JsonRpcException(Server::RPC_INVALID_PARAMETER,
+                                      "To Addr checksum wrong");
     }
   } else {
     LOG_GENERAL(INFO, "Json Data Object has missing components");
+    throw jsonrpc::JsonRpcException(Server::RPC_INVALID_PARAMETER,
+                                    "Json Data Object malformed");
```
Comment (evesnow91) [inline src/libServer/JSONConversion.cpp]: "Json Data Object malformed" -> "Missing components in Json Data Object"

File: src/libServer/JSONConversion.cpp [hunk 3]
```diff
@@ -196,42 +199,51 @@ bool JSONConversion::checkJsonTx(const Json::Value& _json) {
...
       LOG_GENERAL(INFO, "Fault in version");
-      return false;
+      throw jsonrpc::JsonRpcException(Server::RPC_INVALID_PARAMETER,
+                                      "Version not integral");
     }
     if (_json["pubKey"].asString().size() != PUB_KEY_SIZE * 2) {
       LOG_GENERAL(INFO,
                   "PubKey size wrong " << _json["pubKey"].asString().size());
-      return false;
+      throw jsonrpc::JsonRpcException(Server::RPC_INVALID_PARAMETER,
```
Comment (bb111189) [inline src/libServer/JSONConversion.cpp]: RPC_INVALID_ADDRESS_OR_KEY?

File: src/libServer/JSONConversion.cpp [hunk 4]
```diff
@@ -156,15 +157,17 @@ const Transaction JSONConversion::convertJsontoTx(const Json::Value& _json) {
   bytes pubKey_ser;
   if (!DataConversion::HexStrToUint8Vec(pubKey_str, pubKey_ser)) {
     LOG_GENERAL(WARNING, "json cointaining invalid hex str for pubkey");
-    return Transaction();
+    throw jsonrpc::JsonRpcException(Server::RPC_INVALID_PARAMETER,
```
Comment (bb111189) [inline src/libServer/JSONConversion.cpp]: RPC_INVALID_ADDRESS_OR_KEY?

File: src/libServer/JSONConversion.cpp [hunk 5]
```diff
@@ -139,7 +139,8 @@ const Transaction JSONConversion::convertJsontoTx(const Json::Value& _json) {
   bytes toAddr_ser;
   if (!DataConversion::HexStrToUint8Vec(lower_case_addr, toAddr_ser)) {
     LOG_GENERAL(WARNING, "json cointaining invalid hex str for toAddr");
-    return Transaction();
+    throw jsonrpc::JsonRpcException(Server::RPC_INVALID_PARAMETER,
```
Comment (bb111189) [inline src/libServer/JSONConversion.cpp]: RPC_INVALID_ADDRESS_OR_KEY?

File: src/libServer/JSONConversion.cpp (hunk 3 above)
Comment (KaustubhShamshery) [reply src/libServer/JSONConversion.cpp]: The suggestion from dev is that have same code if the issue is in the JSON

Comment (bb111189) [reply src/libServer/JSONConversion.cpp]: I see. Then it is fine.

**Pull request merged**